### PR TITLE
[GITHUB-43] Update to Latest Standards

### PR DESCRIPTION
**Pull request by dennisconrad**
* Migrate to molecule testing
* Flatten variables
* Supersedes #35

### CODE REVIEW

File: defaults/main.yml [hunk 1]
```diff
@@ -1,56 +1,45 @@
...
+  - python-pip
+  - tree
+  - unzip
+sansible_gocd_server_passwd_users:
+  backup: "{SHA}iRIdyZx9uc4lU6CToqsp4H99808="
+sansible_gocd_server_plugins: []
+sansible_gocd_server_port: 8153
+sansible_gocd_server_port_ssl: 8154
+sansible_gocd_server_repo_key_id: 8816C449
+sansible_gocd_server_repo_source: deb https://download.gocd.org /
```
Comment (lobsterdore) [inline defaults/main.yml:41]: 👍

File: meta/main.yml [hunk 2]
```diff
@@ -1,30 +1,25 @@
...
     - development
     - cd
 
 dependencies:
   - role: sansible.users_and_groups
-    users_and_groups:
-      users:
-        - name: "{{ gocd_server.user }}"
-          gecos: Go CD user
+    sansible_users_and_groups_users:
```
Comment (lobsterdore) [inline meta/main.yml]: Can we add group creation as well:

```YAML
 sansible_users_and_groups_groups:
  - name: "{{ sansible_gocd_server_group }}"
 sansible_users_and_groups_users:
  - name: "{{ sansible_gocd_server_user }}"
    group: "{{ sansible_gocd_server_group }}"

File: tasks/build.yml [hunk 3]
```diff
@@ -62,12 +64,12 @@
...
-  become_user: "{{ gocd_server.user }}"
+  become_user: "{{ sansible_gocd_server_user }}"
   get_url:
-    dest: /home/{{ gocd_server.user }}/work/go-server/plugins/external
-    url:  "{{ item }}"
-  with_items: "{{ gocd_server.plugins }}"
+    dest: "/home/{{ sansible_gocd_server_user }}/work/go-server/plugins/external"
+    url: "{{ item }}"
+  with_items: "{{ sansible_gocd_server_plugins }}"
+  notify: restart go-server
```
Comment (lobsterdore) [inline tasks/build.yml]: We normally use list format for notifies:

```YAML
notify:
  - restart go-server
```

File: meta/main.yml (hunk 2 above)
Comment (dennisconrad) [reply meta/main.yml]: Sure.

File: tasks/build.yml (hunk 3 above)
Comment (dennisconrad) [reply tasks/build.yml]: I have _no_ opinion on this :-)

**Comment by lobsterdore** [comment]
LGTM 👍

**Review by rsgoodall — APPROVED** [review]

**Pull request merged**